InternxtDesktop/Views/Widget/WidgetDeviceSelector.swift

```diff
@@ -6,49 +6,67 @@
 //
 
 import SwiftUI
-
-struct Device: Identifiable {
-    var id = UUID()
-    let name: String
-    let index: Int
-    let isCurrentDevice: Bool
-    let isSelected: Bool
-}
+import InternxtSwiftCore
 
 struct WidgetDeviceSelector: View {
-    private let deviceName = ConfigLoader().getDeviceName()
-    @State private var devices: [Device] = []
-    @State private var selectedIndex = 0
-    
+
+    @StateObject var backupsService: BackupsService
+    @State private var selectedDeviceId = 0
+
     var body: some View {
-        VStack(alignment: .leading, spacing: 0) {
-            ForEach(devices) { item in
-                DeviceItem(deviceName: item.name, isCurrentDevice: item.isCurrentDevice, isSelected: self.selectedIndex == item.index) {
-                    withAnimation {
-                        self.selectedIndex = item.index
+        Group {
+            if backupsService.devices.isEmpty {
+                ProgressView()
+                    .progressViewStyle(CircularProgressViewStyle(tint: .blue))
+                    .scaleEffect(2.0, anchor: .center)
+            } else {
+                VStack(alignment: .leading, spacing: 0) {
+                    ForEach(backupsService.devices) { device in
+                        DeviceItem(
+                            deviceName: device.name ?? "",
+                            bucket: device.bucket ?? "",
+                            isSelected: self.selectedDeviceId == device.id
+                        ) {
+                            withAnimation {
+                                self.selectedDeviceId = device.id
+                            }
+                        }
                     }
                 }
+                .onAppear {
+                    self.selectedDeviceId = backupsService.devices.first?.id ?? 0
+                }
             }
         }
-        .frame(width: 160, alignment: .leading)
         .onAppear {
-            devices = [
-                Device(name: deviceName ?? "", index: 0, isCurrentDevice: true, isSelected: true),
-                Device(name: "Home PC", index: 1, isCurrentDevice: false, isSelected: false),
-                Device(name: "Office server", index: 2, isCurrentDevice: false, isSelected: false)
-            ]
+            Task {
+                await backupsService.loadAllDevices()
+            }
         }
+        .frame(width: 160, alignment: .leading)
     }
 }
 
 struct DeviceItem: View {
     
     @Environment(\.colorScheme) var colorScheme
     var deviceName: String
-    var isCurrentDevice: Bool
+    var bucket: String
     var isSelected: Bool
     var onTap: () -> Void
 
+    private let currentDeviceName = ConfigLoader().getDeviceName()
+    private let decrypt = Decrypt()
+    private let config = ConfigLoader().get()
+
+    var plainDeviceName: String? {
+        return try? decrypt.decrypt(base64String: deviceName, password: "\(config.CRYPTO_SECRET2)-\(self.bucket)")
```

Comment:
this looks like is going to block the main thread, also, I believe this is going to decrypt the name every time we compute this property, which means that for every render, the name will be decrypted.

Instead I suggest to decrypt the names when getting the devices, so this view doesn't know anything related to decryption, and we separate this logic.